doc-content/drools-docs/src/main/asciidoc/DecisionEngine/cep-clock-ref.adoc

```diff
@@ -0,0 +1,81 @@
+[id='cep-clock-ref_{context}']
+
+= Session clock implementations in the {DECISION_ENGINE}
+
+During complex event processing, events in the {DECISION_ENGINE} may have temporal constraints and therefore require a session clock that provides the current time. For example, if a rule needs to determine the average price of a given stock over the last 60 minutes, the {DECISION_ENGINE} must be able to compare the stock price event time stamp with the current time in the session clock.
+
+The {DECISION_ENGINE} supports a real-time clock and a pseudo clock. The {DECISION_ENGINE} can simultaneously run multiple event scenarios that may require different clocks.
```

Comment:
This sentence is a bit misleading. You cannot run simultaneously with different clock types in the same session (and it wouldn't make sense to do so).

Reply:
Done. I just deleted that sentence and brought up the next one:

>The {DECISION_ENGINE} supports a real-time clock and a pseudo clock. You can use one or both clock types depending on the scenario: